reproducibility_project/src/utils/trajectory_conversion.py

```diff
@@ -0,0 +1,74 @@
+"""Convert Cassandra trajectories to gsd."""
+import io
+from pathlib import Path
+
+import gsd.hoomd
+import mbuild as mb
+import numpy as np
+import parmed
+
+
+def cassandra2gsd(Hpath, xyzpath, gsdpath, species_list):
+    """Convert Cassandra H and xyz files to a gsd trajectory file."""
+    Hpath = Path(Hpath)
+    xyzpath = Path(xyzpath)
+    gsdpath = Path(gsdpath)
+
+    nspecies = len(species_list)
+    nmols_old = np.zeros(nspecies, dtype=int)
+
+    with Hpath.open() as Hfile, xyzpath.open() as xyzfile, gsd.hoomd.open(
+        str(gsdpath), "wb"
+    ) as gsdfile:
```

Comment:
I'm not sure why these are being converted to Path objects before passing to the context manager.
```
with open(xyzpath) as xyzfile, ...
``` 
works just as well without it

Reply:
Because that way it will take Path objects or strings.

Reply:
It takes any path-like object this way.